togetherly/lib/providers/person_provider.dart

```diff
@@ -0,0 +1,95 @@
+import 'dart:developer';
+
+import 'package:flutter/foundation.dart';
+import 'package:togetherly/models/child.dart';
+import 'package:togetherly/models/parent.dart';
+import 'package:togetherly/models/person.dart';
+import 'package:togetherly/providers/user_identity_provider.dart';
+import 'package:togetherly/services/person_service.dart';
+
+class PersonProvider with ChangeNotifier {
+  PersonProvider(this._service, this._userIdentityProvider) {
+    log("PersonProvider created");
+    refresh();
+  }
+
+  final PersonService _service;
+  UserIdentityProvider _userIdentityProvider;
+
+  List<Parent> _parentList = [];
+  List<Parent> get parentList => _parentList;
+  List<Child> _childList = [];
+  List<Child> get childList => _childList;
+
+  Future<void> addPerson(Person person) async {
+    final familyId = _userIdentityProvider.familyId;
+    if (familyId != null) {
+      person.isParent
+          ? () async {
+              final Parent parent = Parent(
```

Comment:
Once you replace `.isParent` with `is Parent`, constructing a new object in this way will be unnecessary. You should be able to just pass `person` as-is.

Reply:
Should we change what the service is receiving then? Because it wants a Parent or Child, and I have a Person here.

Reply:
I *think* that using a check of `is Parent` will let the compiler know that the object is a Parent, allowing you to pass it as such to the service. I will check though. It is possible that at least an explicit cast is required.

Reply:
I've pushed a change to address this.